sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -51,6 +52,21 @@ public virtual bool IsRetriable(HttpMessage message, Exception exception)
                    (exception is OperationCanceledException && !message.CancellationToken.IsCancellationRequested);
         }
 
+        internal bool IsError(HttpMessage message)
```

Comment:
Why do we need this? Why not call/use IsErrorResponse?

Reply:
I see what you're getting at, it doesn't look like we need it now, does it?  If we go back to just IsErrorResponse, does it even make sense for ResponseClassifier to inherit from MessageClassifier?  Why would it need TryClassify in that case?  (That's why IsError() was there originally, to have a place to test the TryClassify case... that's all moved to StatusCodeClassifier now, though.)

Reply:
It's a good question. The only reason I can think of is that this inheritance hierarchy would allow users to use existing classifiers to override the pipeline's default, i.e. let's say they have a classifier they use in their own code. Now they want to use the exact same classifier to override behavior of somebody else's code/API.

But maybe the scenario is too weak and not worth the complexity.

Reply:
I see, so you're saying they can pass an existing RequestClassifier as a MessageClassifier, and have the TryClassifier call through to IsErrorResponse?

Reply:
But ... I think they can't do that because we're sealing TryClassify, so they can't override it...